src/main/java/store/controller/BuyController.java

```diff
@@ -0,0 +1,92 @@
+package store.controller;
+
+import store.model.Product;
+import store.model.Receipt;
+import store.model.Store;
+import store.view.InputView;
+import store.view.OutputView;
+
+import java.util.Map;
+
+public class BuyController {
+    private final Store store;
+
+    public BuyController(Store store) {
+        this.store = store;
+    }
+
+    public void buyProduct() {
+        boolean continueShopping = true;
+        while (continueShopping) {
+            Receipt receipt = new Receipt();
+            try {
+                Map<String, Integer> purchasedItems = InputView.purchasedItems();
+                processPurchasedItems(receipt, purchasedItems);
+                applyMembershipDiscount(receipt);
+                receipt.print();
+                OutputView.askForAdditionalPurchase();
+                continueShopping = InputView.yOrN();
+                if (continueShopping) OutputView.printInventoryInformation(store);
```

Comment:
아래처럼 메서드로 분리하면 buyProduct() 메서드 내부 흐름이 더 명확히 보일 것 같아요!

```suggestion
                askAdditionalPurchase();
```